support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/support/oauth/authenticator/OAuth20ClientIdClientSecretAuthenticator.java

```diff
@@ -29,12 +30,34 @@
  * @since 5.0.0
  */
 @Slf4j
-@RequiredArgsConstructor
 public class OAuth20ClientIdClientSecretAuthenticator implements Authenticator<UsernamePasswordCredentials> {
+
+    /**
+     * {@link TicketRegistry} for storing and retrieving tickets as needed.
+     */
+    protected final TicketRegistry ticketRegistry;
+
+    /**
+     * The ticket cipher, if any.
+     */
+    protected final CipherExecutor<Serializable, String> registeredServiceCipherExecutor;
+
     private final ServicesManager servicesManager;
     private final ServiceFactory<WebApplicationService> webApplicationServiceServiceFactory;
     private final AuditableExecution registeredServiceAccessStrategyEnforcer;
-    private final CipherExecutor<Serializable, String> registeredServiceCipherExecutor;
+
+    public OAuth20ClientIdClientSecretAuthenticator(final ServicesManager servicesManager,
```

Comment:
Use @RequiredArgsConstructor and delete all params.

Reply:
Due to `checkstyle`, had to move `protected` variables above `private`. Then `@ RequiredArgsConstructor` changes the order of the supplied parameters. To avoid that, ended up declaring constructor. Please let me know if this needs changing, which means fixing the order of constructor param wherever it is used.

Reply:
You can add @Getter and keep everything as final/private, or simply mark everything as `protected`; In this case, it wouldn't hurt much.